src/main/java/lv/ctco/javaschool/goal/boundary/GoalApi.java

```diff
@@ -134,13 +136,17 @@ public boolean isCurrentUsersGoal(@PathParam("id") Long goalId) {
     @Path("/{id}/edit")
     public void editGoal(@PathParam("id") Long goalId, GoalFormDto newGoalDto) {
         User user = userStore.getCurrentUser();
-        if (newGoalDto.getGoalMessage() != null && newGoalDto.getDeadline() != null) {
-            Optional<Goal> goal = goalStore.getUserGoalById(user, goalId);
+        if (newGoalDto.getGoalMessage() != null
+                && newGoalDto.getDeadline() != null
+                && newGoalDto.getDeadline().isAfter(LocalDate.now())) {
+            Optional<Goal> goal = goalStore.getUnachievedUserGoalById(user, goalId);
             goal.ifPresent(g -> {
                 g.setGoalMessage(newGoalDto.getGoalMessage());
                 g.setDeadlineDate(newGoalDto.getDeadline());
+                if (g.getStatus().equals(GoalStatus.OVERDUE))
+                    g.setStatus(GoalStatus.OPEN);
             });
-        }
+        } else throw new ValidationException("Goal cannot be deleted");
```

Comment:
I believe that it is considered a bad practice not to use brackets in if/else statements even if it's one line

Reply:
Thanks, fixed this.